packages/expo-modules-core/src/registerWebModule.ts

```diff
@@ -0,0 +1,12 @@
+import type { NativeModule } from './ts-declarations/NativeModule';
+
+export function registerWebModule<ModuleType extends typeof NativeModule>(
+  moduleName: string,
+  moduleImplementation: ModuleType
+): ModuleType {
+  if (globalThis.expo.modules[moduleName]) {
```

Comment:
We should handle the case when `globalThis.expo` and `globalThis.expo.modules` are not yet available

Reply:
About that:

We would need to have all the classes (NativeModule, EventEmitter) exported to recreate globalThis.expo if it doesn't exist – since we do call the polyfill right above would it be ok to throw here instead if it's not polyfilled?

![image](https://github.com/user-attachments/assets/8ba791f6-daad-4a78-b143-06ba7b915367)

Reply:
Alright, then maybe we should make `expo` optional in the `ExpoGlobal` type?

Reply:
I would agree with @aleqsio. If we don't have a good place to initialize everything, we should create one and ensure that at least the `expo` object is registered. Making it optional postponing the problem.